### DISCUSSION TITLE
How can I pass a value to `helm upgrade --set` for Bitnami PostgreSQL chart

### DISCUSSION THREAD

**Discussion by m90** [helmfile/helmfile#1412] [Questions]
I'm trying to deploy the Bitnami PostgreSQL chart using helmfile.

The initial install works as expected, but on each change / upgrade of my release I will see the following error message:

```
STDERR:
  Error: Failed to render chart: exit status 1: Error: execution error at (superset/charts/postgresql/templates/secrets.yaml:17:24): 
  PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
                   Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
                   Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases
      'global.postgresql.auth.postgresPassword' must not be empty, please add '--set global.postgresql.auth.postgresPassword=$POSTGRES_PASSWORD' to the command. To get the current value:
          export POSTGRES_PASSWORD=$(kubectl get secret --namespace "superset" superset-postgresql -o jsonpath="{.data.postgres-password}" | base64 -d)
  Use --debug flag to render out invalid YAML
  Error: plugin "diff" exited with error
```

When I add the `--set` value to `helmDefaults.args` I will see another error as helmfile tries to call `helm list` which does not know about `--set`.

Is there anyway I can limit `helmDefaults.args` to be used on `update` only (kind of like `diffArgs`)? Or is there any other known workaround for [the issue](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases) when using helmfile?

**Comment by yxxhero** [comment]
@m90  you can try to set it into value file.

**Reply by m90** [reply]
This works but also means (unless I am missing something) I need to hard code my postgres admin password into my values file, which I'd like to avoid.

**Reply by yxxhero** [reply]
You can use environment value.
